src/main/java/com/google/alpollo/AnalysisHelper.java

```diff
@@ -128,4 +139,25 @@ public static String getInterpretation(double score, double magnitude) {
 
     return MIXED;
   }
+
+  public static List<SongEntity> filterDuplicates(List<SongEntity> list) {
+    HashMap<String, SongEntity> map = new HashMap<>(); 
+
+    for (SongEntity entity : list) {
+      if (map.containsKey(entity.getName())) {
+        SongEntity existingEntity = map.get(entity.getName());
+        Set<String> newTypes = entity.getType();
+        Set<String> existingTypes = existingEntity.getType();
+        double newSalience = entity.getSalience() + existingEntity.getSalience();
+        
+        existingTypes.addAll(newTypes);
+        // Apparently 0.84 + 0.06 = 0.899999999, so we have to format the salience again.
+        existingEntity.setSalience(Double.parseDouble(salienceFormat.format(newSalience)));
```

Comment:
I think the setSalience method should be the one responsible for the formatting, as future users might not know they are supposed to pass an already formatted number as argument to the method.
It could even be done in the getSalience method, as this only matters for display

Reply:
I for one think we should not format the salience on the backend at all. Do we have a good reason for it? In my opinion, the frontend should get the salience as it is and format it to 2 decimal places.

Reply:
Yes, I agree with Amaury here actually, it would be much better to have the front end be the one responsible for formatting the data it displays

Reply:
If I don't format the salience in the backend, it's going to be impossible to test. All the salience values will have like 10 decimals and we can find only the first 2. Can I somehow set a tolerance?

Reply:
You can test them for not equality, but almost equality, just set error which will be ok, like 0.01, for example